src/writer/workflows_blocks/airtableintegration.py

```diff
@@ -0,0 +1,352 @@
+                    
+                return self._make_request(method, f"{base_url}/batch", headers, payload)
+            
+        except Exception as e:
+            self.outcome = "error"
+            raise RuntimeError(f"Record operation error: {str(e)}")
+
+    def _handle_tables(self, operation: str, headers: Dict[str, str], base_id: str,
+                      table_name: str = None, data: Dict[str, Any] = None) -> Dict[str, Any]:
+        """Handle table operations"""
+        try:
+            if operation == "list_tables":
+                url = f"{self.METADATA_URL}/bases/{base_id}/tables"
+                return self._make_request("GET", url, headers)
+            
+            elif operation == "get_table_schema":
+                url = f"{self.METADATA_URL}/bases/{base_id}/tables/{quote(table_name)}"
+                return self._make_request("GET", url, headers)
+            
+            elif operation == "create_table":
+                url = f"{self.METADATA_URL}/bases/{base_id}/tables"
+                return self._make_request("POST", url, headers, data)
+            
+            elif operation == "update_table":
+                url = f"{self.METADATA_URL}/bases/{base_id}/tables/{quote(table_name)}"
+                return self._make_request("PATCH", url, headers, data)
+            
+        except Exception as e:
+            self.outcome = "error"
+            raise RuntimeError(f"Table operation error: {str(e)}")
+
+    def _handle_views(self, operation: str, headers: Dict[str, str], base_id: str,
+                     table_name: str, view_name: str = None, 
+                     data: Dict[str, Any] = None) -> Dict[str, Any]:
+        """Handle view operations"""
+        try:
+            base_url = f"{self.METADATA_URL}/bases/{base_id}/tables/{quote(table_name)}/views"
+            
+            if operation == "list_views":
+                return self._make_request("GET", base_url, headers)
+            
+            elif operation == "get_view":
+                return self._make_request("GET", f"{base_url}/{quote(view_name)}", headers)
+            
+            elif operation == "create_view":
+                return self._make_request("POST", base_url, headers, data)
+            
+        except Exception as e:
+            self.outcome = "error"
+            raise RuntimeError(f"View operation error: {str(e)}")
+
+    def run(self):
+        try:
+            # Get authentication and base configuration
+            api_key = self._get_field("api_key")
+            base_id = self._get_field("base_id")
+            headers = self._get_headers(api_key)
+
+            # Get operation details
+            operation = self._get_field("operation")
+            table_name = self._get_field("table_name", True)
+            record_id = self._get_field("record_id", True)
+            view_name = self._get_field("view_name", True)
+            data = json.loads(self._get_field("data", True, "{}"))
```

Comment:
Some fields are not JSON, and `data` is parsed twice
```suggestion
            table_name = self._get_field("table_name")
            record_id = self._get_field("record_id")
            view_name = self._get_field("view_name")
            data = self._get_field("data", True, "{}")
```